packages/grafana-data/src/types/dataLink.ts

```diff
@@ -1,11 +1,32 @@
+import { ScopedVars } from './ScopedVars';
+
+/**
+ * Callback info for DataLink click events
+ */
+export interface DataLinkClickEvent<T = any> {
+  origin: T;
+  scopedVars: ScopedVars;
+  e?: any; // mouse/react event?
+}
+
 /**
  * Link configuration.  The values may contain variables that need to be
  * processed before running
  */
 export interface DataLink {
-  url: string;
   title: string;
   targetBlank?: boolean;
+
+  // 3: The URL if others did not set it first
+  url: string;
+
+  // 2: If exists, use this to construct the URL
+  // Not saved in JSON/DTO
+  onBuildHref?: (event: DataLinkClickEvent) => string;
```

Comment:
Just wondering if this is necessary, as this could be handled in `onClick` handler that would redirect accordingly if necessary. But I don't have strong opinion on that. Maybe instead of introducing another property to this interface we could have `url` prop accept either string or function? Not the cleanest solution but doesn't make the API more complex.

Reply:
I added it to start the conversation :)  The difference (I think) is that the 'href' element gets set, so things like ctrl+click open in a new window.

BUT, we should probably have fewer options and only support `onClick`

Reply:
Yeah, I prefer built in browser behaviours too rather than handling the navigation programatically. Maybe onBuildHref ain't that bad idea.

Reply:
since the other field is named url, onBuildUrl(...) makes more sense